File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/riivolution/ui/RiivolutionItem.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.riivolution.ui;

public class RiivolutionItem
{
  public final int mDiscIndex;
  public final int mSectionIndex;
  public final int mOptionIndex;

  /**
   * Constructor for a disc.
   */
  public RiivolutionItem(int discIndex)
  {
    mDiscIndex = discIndex;
    mSectionIndex = -1;
    mOptionIndex = -1;
  }

  /**
   * Constructor for a section.
   */
  public RiivolutionItem(int discIndex, int sectionIndex)
  {
    mDiscIndex = discIndex;
    mSectionIndex = sectionIndex;
    mOptionIndex = -1;
  }

  /**
   * Constructor for an option.
   */
  public RiivolutionItem(int discIndex, int sectionIndex, int optionIndex)
  {
    mDiscIndex = discIndex;
    mSectionIndex = sectionIndex;
    mOptionIndex = optionIndex;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/riivolution/ui/RiivolutionItem.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.riivolution.ui

class RiivolutionItem {
    val mDiscIndex: Int
    val mSectionIndex: Int
    val mOptionIndex: Int

    /**
     * Constructor for a disc.
     */
    constructor(discIndex: Int) {
        mDiscIndex = discIndex
        mSectionIndex = -1
        mOptionIndex = -1
    }

    /**
     * Constructor for a section.
     */
    constructor(discIndex: Int, sectionIndex: Int) {
        mDiscIndex = discIndex
        mSectionIndex = sectionIndex
        mOptionIndex = -1
    }

    /**
     * Constructor for an option.
     */
    constructor(discIndex: Int, sectionIndex: Int, optionIndex: Int) {
        mDiscIndex = discIndex
        mSectionIndex = sectionIndex
        mOptionIndex = optionIndex
    }
}
